packages/transport-websockets/src/listener.ts

```diff
@@ -4,40 +4,69 @@ import { ipPortToMultiaddr as toMultiaddr } from '@libp2p/utils/ip-port-to-multi
 import { multiaddr, protocols } from '@multiformats/multiaddr'
 import { createServer } from 'it-ws/server'
 import { socketToMaConn } from './socket-to-conn.js'
-import type { ComponentLogger, Logger, Connection, Listener, ListenerEvents, CreateListenerOptions } from '@libp2p/interface'
+import type { ComponentLogger, Logger, Connection, Listener, ListenerEvents, CreateListenerOptions, CounterGroup, MetricGroup, Metrics } from '@libp2p/interface'
 import type { Multiaddr } from '@multiformats/multiaddr'
 import type { Server } from 'http'
 import type { DuplexWebSocket } from 'it-ws/duplex'
 import type { WebSocketServer } from 'it-ws/server'
 
 export interface WebSocketListenerComponents {
   logger: ComponentLogger
+  metrics?: Metrics
 }
 
 export interface WebSocketListenerInit extends CreateListenerOptions {
   server?: Server
 }
 
+export interface WebSocketListenerMetrics {
+  status: MetricGroup
+  errors: CounterGroup
+  events: CounterGroup
+}
+
+enum WebSocketListenerStatusCode {
```

Comment:
Note: I removed the `paused` status that the TCP transport had because it didn't seem to make sense here.

Reply:
Listener status was added to the TCP transport as it can shut itself down when under attack and restart later on.  This feature isn't in the WebSocket transport so I think we don't need this right now.

We may add it later I guess but let's do that when we do that.